wechaty-puppet/helper/async.go

```diff
@@ -22,40 +23,44 @@ type (
 	}
 
 	async struct {
-		wg        sync.WaitGroup
-		tasks     []Task
-		workerNum int
+		tasks        []Task
+		maxWorkerNum int
 	}
 
 	// Task task func
 	Task func() (interface{}, error)
 )
 
-func NewAsync(workerNum int) IAsync {
-	if workerNum <= 0 {
-		workerNum = DefaultWorkerNum
+// NewAsync ...
+func NewAsync(maxWorkerNum int) IAsync {
+	if maxWorkerNum <= 0 {
+		maxWorkerNum = DefaultWorkerNum
 	}
 	return &async{
-		wg:        sync.WaitGroup{},
-		workerNum: workerNum,
+		maxWorkerNum: maxWorkerNum,
 	}
 }
 
 func (a *async) AddTask(task Task) {
-	a.wg.Add(1)
 	a.tasks = append(a.tasks, task)
 }
 
 func (a *async) Result() []AsyncResult {
 	taskChan := make(chan Task)
 	resultChan := make(chan AsyncResult)
-	for i := 0; i < a.workerNum; i++ {
+
+	taskNum := len(a.tasks)
+	wg := sync.WaitGroup{}
```

Comment:
Change `wg` to local variable, when I reuse async, wg will be recreated, will there be performance loss?

Reply:
That make sense. so I change it back.